documentation/sdkautomation/sdk_customization.md

```diff
@@ -0,0 +1,762 @@
+        // installInstructionInput.json: See #InstallInstructionScriptInput .
+        // installInstructionOutput.md: Install instruction in markdown.
+        "allOf": {
+          "$ref": "#/definitions/RunOptions"
+        },
+        "properties": {
+          "enableLiteInstallInstruction": {
+            "type": "boolean",
+            "default": false
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RunOptions": {
+      // Options to run a script and collect log.
+      "type": "object",
+      "properties": {
+        "path": {
+          // Script path related to repo root
+          "type": "string"
+        },
+        "logPrefix": {
+          // Prefix to be added to SDK Automation log. By default it would be filename of the script.
+          "type": "string"
+        },
+        "stdout": {
+          // How should SDK Automation handle the script stdout stream
+          "$ref": "#/definitions/RunLogOptions"
+        },
+        "stderr": {
+          // How should SDK Automation handle the script stderr stream
+          "$ref": "#/definitions/RunLogOptions"
+        },
+        "exitCode": {
+          // How should SDK Automation handle non-zero exitCode.
+          "showInComment": {
+            // Should we show this error in comment.
+            "type": "boolean",
+            "default": true
+          },
+          "result": {
+            // If script has non-error exitCode how should we mark the script's result.
+            "type": "string",
+            "enum": [
+              "error", "warning", "ignore"
+            ],
+            "default": "error"
+          }
+        },
+        "required": [
+          "path"
+        ]
+      }
+    },
+    "RunLogOptions": {
+      // How should SDK Automation handle the log stream.
+      "showInComment": {
+        // Should we show this stream in comment.
+        "$ref": "#/definitions/RunLogFilterOptions"
+      },
+      "scriptError": {
+        // If any line match, assume the script fails.
+        "$ref": "#/definitions/RunLogFilterOptions"
+      },
+      "scriptWarning": {
+        // If any line match, assume the script warns.
+        "$ref": "#/definitions/RunLogFilterOptions"
+      }
+    },
+    "RunLogFilterOptions": {
+      "oneOf": [
+        {
+          // If line of log match this regex then hit
+          "type": "string",
+          "format": "regex"
+        },
+        {
+          // If set to true, any line of log will hit
+          "type": "boolean"
+        }
+      ],
+      "default": false
+    }
+  }
+}
+```
+
+### GenerateInput
+
+Input file for generate script.
+
+#### GenerateInput Example
+
+```jsonc
+{
+  "specFolder": "/z/work/azure-rest-api-specs",
+  "headSha": "fce3400431eff281bddd04bed9727e63765b8da0",
+  "headRef": "refs/pull/1234/merge",
+  "repoGitUrl": "git@github.com:Azure/azure-rest-api-specs.git",
+  "repoHttpsUrl": "https://github.com/Azure/azure-rest-api-specs.git",
```

Comment:
I don't think it necessary to provide these duplicated values it is just another place something could be out of sync, so if there is no scenario for them both I would just keep the repoUrl as the https url.

Reply:
OK. Remove repoGitUrl